packages/flutter_tools/lib/src/cache.dart

```diff
@@ -221,6 +235,27 @@ class Cache {
     return getCacheArtifacts().childDirectory(name);
   }
 
+  /// Get a file in a cached artifact directory; for example:
+  ///
+  /// `getArtifactFileName('libimobiledevice', 'idevice_id');`
+  ///
+  /// will return: `'bin/cache/artifacts/libimobiledevice/idevice_id'`
+  String getArtifactFile(String dirName, String fileName) {
+    return fs.path.join(getArtifactDirectory(dirName).path, fileName);
+  }
+
+  String get iosUsbExecutionPath {
```

Comment:
Is there any way to make this a bit less platform specific. For example, we might run into a similar situation with artifacts on Linux. Maybe change this to `artifactExecutionEnv` and have it return a `Map<String, String>`. It would iterate over `_artifacts`, and ask each of them if it wants to add anything to the environment. Then you'd cache the result so it doesn't have to be recomputed each time.

Reply:
done